### ISSUE TITLE
"Corrupted" values on load and save

### ISSUE THREAD

**Issue by griggsrl** [x-stream/xstream#8]
My team has recently inherited a codebase that utilizes XStream 1.4.7 to load and save XML for configuration settings which then de-/serializes them from/to custom POCOs. The problem that we're having is that some of the values are getting corrupted during reads or writes. It isn't consistently occurring either which makes it that much more unusual. In most cases, it works perfectly fine with the exact same XML and the exact same POCOs.

A very simplified example (I can't post the exact code and it's a bit more complex so I'm going for an easy way to explain what we're seeing) is given the XML:

    <monitor>
      <autostart>true</autostart>
      <name>MYVALUE</name>
    </monitor>

Mapped to a POCO:

    public class MonitorEntry {
        public Boolean autostart;
        public String name;
    }

Loaded with XStream:

    XStream xStream = new XStream(new DomDriver());
    xStream.alias("Monitor", MonitorEntry.class);
    Monitor monitor = (Monitor)xStream.fromXML(myFile);

The value of name in the Monitor object is read in as `arVALUE` instead of `MYVALUE`. The garbage characters at the beginning are what throws things off. Even more strangely, if we change the value of the `<autostart>` element to false then the XML is mapped correctly and the garbage characters do not appear.

To add to the mystery, on our end we're only seeing the corruption on loading XML to objects, but on one particular customer system they are seeing corruption only when actually saving the XML from objects. In this case, it's exactly the opposite of the above scenario. Given the same POCO with name set to `MYVALUE`, the actual XML written to the XML file becomes:

    <monitor>
        <autostart>true</autostart>
        <name>arVALUE</name>
    </monitor>

Now for a string value such as name it isn't much of a functional issue as it's just a name that is then just spelled wrong but where this becomes a problem is when mapping the XML value to, for example, an enum and the mapping can't find the enum value.

An example being if there is an enum:

    public enum Type { VALUE1, VALUE2 };

And the POCO is:

    public class MonitorEntry {
        public Boolean autostart;
        public String name;
        public Type type;
    }

With the XML:

    <monitor>
        <autostart>true</autostart>
        <name>MYNAME</name>
        <type>VALUE2</type>
    </monitor>

But the XML value is being read by XStream as `erLUE2` then the XStream mapping won't be able to match the correct enum value and throws an exception such as:

    No enum const class com.sample.MonitorEntry$Type.erLUE2

We tried updating to XStream 1.4.8 just to see if perhaps something had been fixed but the behavior persists. The codebase is set to compile to Java 1.6 but we've tried 1.6, 7, and 8 as runtimes just to see if it was a JRE difference or something else environmental.

Has anyone else seen similar issues or have any suggestions on what might cause this? I can further update my post to include more detail if necessary. We've used XStream quite a bit before but never had issues so this is a bit unusual.

**Comment by joehni** [comment]
Hi, sorry, for the delay, but I was 3 weeks offline. However, I've never seen such a behavior before and I have no idea, why XStream should behave in this way. Can you try to use the Xpp3 parser instead? XStream simply processes what is delivered by the parser, so I'd like to know, if that makes a change. Do you use custom converters or mappers?

**Comment by griggsrl** [comment]
Not a problem. :) 

None of us have seen this behavior before either, it's definitely very strange. I'll try the default Xpp3 parser when I get a chance but we did try the StaxDriver and that resulted in a NullPointerException that didn't have any additional info in it which was unhelpful.

Currently we don't have any custom converters or mappers so it's pretty simple stuff.

One other thing to note that we found odd in this whole process is the customer that was getting the corrupted values when writing to a file installed our software on a brand new machine (same OS version, save version of Java) as a test and have had no issues with it since. We're still seeing the issue with loading on some of our dev machines also, so it could possibly be purely environmental.
